deepchecks/vision/checks/performance/robustness_report.py

```diff
@@ -0,0 +1,244 @@
+
+PR = TypeVar('PR', bound='RobustnessReport')
+
+
+class RobustnessReport(SingleDatasetCheck):
+    """Check several image enhancements for model robustness.
+
+    Parameters
+    ----------
+        alternative_metrics (List[Metric], default None):
+            A list of ignite.Metric objects whose score should be used. If None are given, use the default metrics.
+    """
+
+    def __init__(self,
+                 alternative_metrics: Optional[List[Metric]] = None,
+                 prediction_extract: Optional[Callable] = None,
+                 augmentations=None,
+                 epsilon: float = 10 ** -2,
+                 random_seed: int = 42):
+        super().__init__()
+        self._epsilon = epsilon
+        self.alternative_metrics = alternative_metrics
+        self.prediction_extract = prediction_extract
+        self.random_seed = random_seed
+        self.augmentations = augmentations
+
+    def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResult:
+        """Run check.
+
+        Returns
+        -------
+            CheckResult: value is dictionary in format 'score-name': score-value
+        """
+        self._set_seeds(self.random_seed)
+        context.assert_task_type(TaskType.CLASSIFICATION, TaskType.OBJECT_DETECTION)
+        if dataset_type == 'train':
+            dataset = context.train
+        else:
+            dataset = context.test
+
+        transform_type = dataset.transform_type
+        task_type = dataset.task_type
+        model = context.model
+        self.augmentations = self.augmentations or get_robustness_augmentations(transform_type)
+        # Get default scorers if no alternative, or validate alternatives
+        # scorers = get_scorers_list(model, dataset, dataset.get_num_classes(), self.alternative_metrics)
+        scorers = []
+
+        results_df, examples = self._report_inner_loop(dataset, model, scorers)
+
+        bad_example_figures = self._create_example_figure(examples)
+
+        # TODO visualiztion
+        plot_x_axis = 'Class'
+        fig = px.histogram(
+            results_df,
+            x=plot_x_axis,
+            y='Value',
+            color='Dataset',
+            barmode='group',
+            facet_col='Metric',
+            facet_col_spacing=0.05,
+            hover_data=['Number of samples']
+        )
+
+        if task_type == TaskType.CLASSIFICATION:
+            fig.update_xaxes(tickprefix='Class ', tickangle=60)
+
+        fig = (
+            fig.update_xaxes(title=None, type='category')
+                .update_yaxes(title=None, matches=None)
+                .for_each_annotation(lambda a: a.update(text=a.text.split('=')[-1]))
+                .for_each_yaxis(lambda yaxis: yaxis.update(showticklabels=True))
+        )
+
+        return CheckResult(
+            results_df,
+            header='Robustness Report',
+            display=fig
+        )
+
+    def _report_inner_loop(self, dataset: VisionDataset, model, scorers):
```

Comment:
Think we want to call metrics metrics here, as that is their ignite name. @ItayGabbay WDYT?